### PR TITLE
DB Schema migration for module update

### PR DESCRIPTION
**Pull request by TarasSpashchenko**
FOLIO-2169 DB Schema migration for module update (DB-schema-migration.md) added

This documentation includes information about possible strategies regarding DB schema migration that can be used under a module's version update.

### CODE REVIEW

File: DB-schema-migration.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
+# DB Schema migration for module update
+
+Copyright (C) 2016-2019 The Open Library Foundation
```
Comment (julianladisch) [inline DB-schema-migration.md]: Can you remove the Copyright line? The line in the README.md covers the whole repository and is the single place where we want to update the year.
See https://dev.folio.org/guidelines/contributing/#no-license-header

File: DB-schema-migration.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+This software is distributed under the terms of the Apache License, Version 2.0.
+See the file ["LICENSE"](LICENSE) for more information.
```
Comment (julianladisch) [inline DB-schema-migration.md]: Can you remove the license information? The LICENSE file covers the whole repository.
See https://dev.folio.org/guidelines/contributing/#no-license-header

File: DB-schema-migration.md (hunk 1 above)
Comment (TarasSpashchenko) [reply DB-schema-migration.md]: Done

File: DB-schema-migration.md (hunk 2 above)
Comment (TarasSpashchenko) [reply DB-schema-migration.md]: Done

**Comment by julianladisch** [comment]
Can you add an example of a schema.json table section with multiple changes? For example where the JSON property foo is added in version 2.0.0, foo is renamed to bar in version 3.0.0, and an index for bar is added in version 3.1.0?

File: DB-schema-migration.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+* The current version of a module enabled for a tenant is mod-inventory-storage-1.0.0-SNAPSHOT.274
+* The version of a module that will be enabled for a tenant (it is provided in the ModuleDescriptor.json) is mod-inventory-storage-1.0.0-SNAPSHOT.275
+* The version specified in the “fromModuleVersion” in schema.json is mod-inventory-storage-1.0.0
+
+The comparison will be done between ```mod-inventory-storage-1.0.0``` and ```mod-inventory-storage-1.0.0-SNAPSHOT.274```
+
+The result is ```mod-inventory-storage-1.0.0 > mod-inventory-storage-1.0.0-SNAPSHOT.274``` so a script/table will be applied/created or changed as well.
+
+### DoD statement for DB schema migration
```
[the remark this answers is not in the document]
Comment (TarasSpashchenko) [reply DB-schema-migration.md]: Done

File: DB-schema-migration.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+3. Deploy a new version of the module using POST /_/discovery/modules
+4. A Tenant decides to update the module’s version.
+5. A maintenance period must be started here so users can't work with the FOLIO applications that uses this module.
+6. Create a dump of the DB schema used by this module for a tenant.
+5. Install the new version of the module for a tenant using POST /_/proxy/tenants/{tenant}/install
+6. Check the log of the module for any errors or issues.
+
+All the use cases for DB changes remain valid and must be taken into consideration except that compensation scripts are not needed in this scenario.
+
+In case of any issues or errors during a module’s update, all we need to do is to restore the schema's backup and  enable the previous version of the module for the tenant.
```
[the remark this answers is not in the document]
Comment (TarasSpashchenko) [reply DB-schema-migration.md]: Done

File: DB-schema-migration.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    - Certainly, the simplest option here is to perform module's version update at the maintenance period when tenant's users do not work with the platform. It eliminates all these additional tricky actions necessary not to lose data
+
+Certainly, all the use cases for DB changes remain valid and must be taken into consideration except that compensation scripts are not needed in this scenario.
+
+In case of any issues or errors during a module’s update, all we need to do is to enable the previous version of the module for the tenant. The only drawback here is that if we find any problems after some time when users have already worked with the new version of the module we also need to provide backward replication of data from the DB schema of the new version to the DB schema of the old one.
```
[the remark this answers is not in the document]
Comment (TarasSpashchenko) [reply DB-schema-migration.md]: Done

**Pull request merged**